plugin-core/src/main/java/appland/cli/DefaultCommandLineService.java

```diff
@@ -572,7 +577,57 @@ private static GeneralCommandLine applyServiceEnvironment(@NotNull GeneralComman
         return commandLine.withParentEnvironmentType(environmentType)
                 .withEnvironment(settings.getCliEnvironment())
                 .withEnvironment(isNotEmpty(appMapKey) ? Map.of("APPMAP_API_KEY", appMapKey) : Map.of())
-                .withEnvironment(isNotEmpty(openAIKey) ? Map.of("OPENAI_API_KEY", openAIKey) : Map.of());
+                .withEnvironment(isNotEmpty(openAIKey) ? Map.of("OPENAI_API_KEY", openAIKey) : Map.of())
+                .withEnvironment(createProxyEnvironment());
+    }
+
+    /**
+     * Creates a map of proxy environment variable settings.
+     * The following variables are supported:
+     * - http_proxy
+     * - https_proxy
+     * - no_proxy
+     * <p>
+     * SOCKS proxy settings of the IDE are not supported. In this case, an empty map is returned.
+     * <p>
+     * Autoconfigured PAC proxy settings are not supported. In this case, an empty map is returned.
+     * <p>
+     * If the settings define proxy authentication settings, then username and password are part of the proxy URLs.
+     *
+     * @return Map of UNIX-style proxy environment variables, based on the IDE's settings.
+     * The map contains lower-case and upper-case variables, e.g. http_proxy and also HTTP_PROXY.
+     * If the IDE does not define proxy settings, then an empty map is returned.
+     */
+    static @NotNull Map<String, String> createProxyEnvironment() {
+        var settings = HttpConfigurable.getInstance();
+        if (!settings.USE_HTTP_PROXY || settings.PROXY_TYPE_IS_SOCKS) {
+            return Map.of();
+        }
+
+        // we're following the SDK's com.intellij.util.net.HttpConfigurable.getJvmProperties
+        var proxyLogin = settings.getProxyLogin();
+        var usernameAndPassword = settings.KEEP_PROXY_PASSWORD && StringUtil.isNotEmpty(proxyLogin)
+                ? String.format("%s:%s", proxyLogin, settings.getPlainProxyPassword())
+                : null;
+
+        try {
+            var url = new URI("http", usernameAndPassword, settings.PROXY_HOST, settings.PROXY_PORT, null, null, null).toURL();
+            var proxyURL = url.toExternalForm();
+            var noProxy = settings.PROXY_EXCEPTIONS;
+            return Map.of(
+                    // lower-cased names
+                    "http_proxy", proxyURL,
+                    "https_proxy", proxyURL,
+                    "no_proxy", noProxy,
+                    // upper-cased names
+                    "HTTP_PROXY", proxyURL,
+                    "HTTPS_PROXY", proxyURL,
+                    "NO_PROXY", noProxy
```

Comment:
It's fine to do just upper or lower case, we will check for both in the RPC service.